test/langtools/tools/javac/api/8198317/T8198317.java

```diff
@@ -58,17 +66,61 @@ public static void main(String[] args) throws Exception {
 
     @Test
     public void testLogSettingInJavacTool() throws Exception {
-        // TODO Situation: out is null and the value is not set in the context.
-        // TODO Situation: out is not null and out is not a PrintWriter.
+        String code = """
+                import java.io.Serializable;
+                class Test implements Serializable {
+                    public static final int serialVersionUID = 1;
+                }""";
+
+        List<String> expected = Arrays.asList(
+                "Test.java:3:29: compiler.warn.long.SVUID: Test",
+                "1 warning");
+
+        List<? extends JavaFileObject> files = Arrays.asList(new MemFile("Test.java", code));
+
+        // Situation: out is null and the value is not set in the context.
+        ByteArrayOutputStream bais = new ByteArrayOutputStream();
+        PrintStream printStream = new PrintStream(bais);
+        PrintStream prev = System.err;
+        System.setErr(printStream);
+        ToolProvider.getSystemJavaCompiler()
+                .getTask(null, null, null, Arrays.asList("-XDrawDiagnostics", "-Xlint:serial"), null, files)
+                .call();
+        tb.checkEqual(expected, Arrays.asList(bais.toString().split(lineSeparator)));
+        System.setErr(prev);
```

Comment:
Although not strictly necessary in this case, it is good practice to use `try ... finally` to reset values back to some previous value. In other words, the `System.setErr(prev);` should be in the `finally` clause.

Reply:
Fixed.